enamel/api/handlers.py

```diff
@@ -45,6 +53,22 @@ def handle_error(error):
     return response
 
 
+def handle_404(error):
+    """Transform a Flask 404 into our error handling flow."""
+    return handle_error(httpexceptor.HTTP404(error.description))
+
+
+def set_request_id():
+    """A before_request function to set required-id."""
+    flask.g.request_id = uuidutils.generate_uuid()
```

Comment:
Me neither, but somewhere along the way we chose Flask, and flask doesn't want to give me access to `environ`. I've been able to contain `app` in a relatively small area, but passing context between various handlers has proven messy.

I'd be _totally_ happy to go frameworkless and just use WSGI-in-the-raw (as God intended!) but then I reckon people who come after us will moan.